docs/compile-time-parameters.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "Compile Time Parameters"
+category: "dsl"
+weight: 10
+anchors:
+  - title: "Usage"
+    url: "#usage"
+  - title: "Template Nesting"
+    url: "#template-nesting"
+---
+
+## Compile Time Parameters
+
+Compile time parameters are parameters utilized during the compilation
+of a template. These parameters can _only_ be used via the SparkleFormation
+library. It is for this reason that compile time parameters are generally
+discouraged from use. In most situations the structure of the template
+can be refactored to remove any requirement of compile time parameters. To
+handle the cases that fall outside of "most situations", SparkleFormation
+provides support for compile time parameters.
+
+### Usage
+
+Compile time parameters are defined during the instantiation of a template:
+
+~~~ruby
+SparkleFormation.new(:test,
+  :compile_time_parameters => {
+    :number_of_nodes => {
+      :type => :number,
+      :default => 1
+    }
+  }
+) do
+
+  state!(:number_of_nodes).times do |i|
+    dynamic!(:ec2_instance, "node_#{i}")
+  end
+
+end
+~~~
+
+#### Parameter Declaration
+
+Declaring compile time parameters is done via the `:compile_time_parameters`
+option when instantiating a new template. This option expects a `Hash` value
+with the compile time parameter names as the key, and the parameter options
```

Comment:
I find this hard to read. Maybe "This option expects a `Hash` value in which the keys are parameter names, and their respective values are a `Hash` defining options for the parameter." ?